pkg/cloudprovider/util/net.go

```diff
@@ -65,3 +65,9 @@ func ContainsCIDR(cidrs []string, version IPVersion) bool {
 
 	return false
 }
+
+// IsLinkLocal checks if given ip address is link local
```

Comment:
Can you please explain more, why do we need only locally linked ips?

Reply:
This is used to filter them out because showing them in the machine status is not useful.

Reply:
ah! understood. :) 
Will add in next PRs. All the e2e tests are green don't want to babysit them right now.